include/ignition/gazebo/Conversions.hh

```diff
@@ -709,6 +709,22 @@ namespace ignition
     /// \return Particle emitter message.
     template<>
     sdf::ParticleEmitter convert(const msgs::ParticleEmitter &_in);
+
+    /// \brief Generic conversion from an SDF element to another type.
+    /// \param[in] _in SDF element.
+    /// \return Conversion result.
+    /// \tparam Out Output type.
+    template<class Out>
+    Out convert(const sdf::Element &/*_in*/)
+    {
+      Out::ConversionNotImplemented;
+    }
+
+    /// \brief Specialized conversion from an SDF element to a plugin message.
+    /// \param[in] _in SDF element.
+    /// \return Plugin message.
+    template<>
+    msgs::Plugin convert(const sdf::Element &_in);
```

Comment:
I used `sdf::Plugin` wherever it was reasonable in bf93928dd3491e766e2fcf161e7bdba32a0a1998.

I still need to use `sdf::Element`, mainly for the `SystemManager`. Element is still widely used there, even after #1352, because that's in the public `System::Configure` API.